docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+
+### Example Specifications
+
+**1. An adopted S3 bucket with the minimal number of fields**
+
+```
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: AdoptedResource
+metadata:
+  name: adopted-data-bucket
+spec:
+   apiVersion: s3.services.k8s.aws/v1alpha1
+   kind: Bucket
+   identifier:
+        arn: arn:aws:s3:::my-data-bucket
+```
+
+**2. An adopted S3 bucket with overriding metadata fields**
+
+```
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: AdoptedResource
+metadata:
+  name: adopted-data-bucket
+spec:
+   apiVersion: s3.services.k8s.aws/v1alpha1
+   kind: Bucket
+   metadata:
+        name: data-bucket
+        namespace: team-a
+        annotations:
+            ...
+   identifier:
+        arn: arn:aws:s3:::my-data-bucket
+```
+
+**3. An adopted API Gateway v2 API (using an ID identifier)**
+
+```
+apiVersion: apigateway.services.k8s.aws/v1alpha1
+kind: AdoptedResource
+metadata:
+  name: adopted-api-gateway
+spec:
+   apiVersion: apigateway.services.k8s.aws/v1alpha1
+   kind: Api
+   identifier:
+        id: 123456789012
+```
+
+### Custom Resource Validation Webhook
+
+In order to minimise the chance of user’s facing unexpected behaviours, a validating webhook will validate any adopted resource changes prior to submission to the ACK managers. The webhook will primarily reject the following error cases:
+
+1. The provided target ACK type does not exist within the cluster
+2. The target resource would override an existing resource
+3. The target resource type cannot be adopted
+
+## Adopted K8s Resource Tagging
+
+All adopted resources will be annotated with an ACK-specific annotation that differentiates it from a natively created resource. This annotation serves to ensure clarity for users about possible specification differences introduced during the adoption process, or for filtering by adopted status. However, once adopted into a cluster, a resource will be reconciled with the same logic as any natively created resource. 
+
+The proposed k8s adopted tag is `services.k8s.aws/adopted: true`
+
+The adopted annotation would remain on the custom resource for the remainder of its lifecycle, and should not be modified by any other manager operation. 
+
+## AWS Resource Tagging
+
+AWS resources adopted by an ACK controller will have tags retroactively applied (whenever possible) to indicate its association with an ACK deployment. The details of these tags are discussed in another issue ([#148](https://github.com/aws/aws-controllers-k8s/issues/148)). ACK managers should not place any conditions on the existence of ACK tags on the AWS resources to gate the adoption process. The manager should should override any existing ACK tags.
```

Comment:
> The manager should override any existing ACK tags

Hm... could the ACK Manager overwriting the tags maybe lead to fights over resources? Say I have an S3 bucket that is managed from ACK running in cluster `one`. I accidentally create an `AdoptedResource` for that same bucket in ACK running in cluster `two`. In that case, will `two` just take over management of that S3 bucket?

Reply:
This is a point we brought up. Ultimately this is the same if the resource is managed by CDK, TF, Cfn, other operator, etc. Without a solid way of locking a resource to a given cluster, there is no way to strictly ensure it is only reconciled by at most one service.
The ACK controller should have the privilege to override a tag, though, as the use case of migration from one cluster to another would require this.